src/replicache.ts

```diff
@@ -595,7 +594,9 @@ export default class Replicache implements ReadTransaction {
 
     if (this._syncPromise !== null) {
       await this._syncPromise;
-      await this.sync();
+      // Call schedule instead of sync to debounce/dedupe multiple calls.
```

Comment:
So now they all queue up same as before, but each one when it runs schedules itself, then the next one de/reschedules, then the next one de/reschedules, etc.

I suppose this works, but if there are many syncs queued (as is common in realtime applications) this can easily cause the next sync to be for dozens of ms, which is not what we want.

We want the pushes to be dispatched as rapidly as possible (we should actually remove `pushDelay` all together I think, or default it very very low, like 5ms).

I think if you want to take something like the approach you have, you could go the other way -- we could track if there's a sync call queued and if there is then subsequent sync calls are dropped.

Reply:
I think you are mistaken. They are all going to get invoked in the same task as `this._syncPromise` resolves so they will do semantically the same as the following:

```
for (let i < 0; i < 100; i++) {
  this._clearTimer()
  this._scheduleSync(1);
}
```

The only extra time accumulated is the time in between the these calls which should be a few µs.

I could change the test to use 100 sync calls if that would make you more comfortable?

Reply:
You're right, my mistake.

I find this logic kind of hard to follow. I feel like solution (1) would be easier to reason about, once push/pull are factored apart. Do you agree or is it fine as-is for you?

Reply:
1 being the solution to check if there are still pending mutations and push again?